app/models/solid_queue/job/clearable.rb

```diff
@@ -11,7 +11,14 @@ module Clearable
 
       class_methods do
         def clear_finished_in_batches(batch_size: 500, finished_before: SolidQueue.clear_finished_jobs_after.ago)
-          clearable(finished_before: finished_before).in_batches(of: batch_size).delete_all
+          loop do
+            records_deleted = \
+              SolidQueue::Job.
+                clearable(finished_before: finished_before).
```

Comment:
`finished_before` is stable: This ensure that the loop will stop after processing al the records created before `finished_before`